### PR TITLE
Code Mozilla Stories Blog

### PR DESCRIPTION
**Pull request by nathan-barrett**
## One-line summary

This PR adds the Mozilla stories blog (without contentful integration)

## Significant changes and points to review
There is a landing page and three different articles - each article will extend the `stories-base-article.html` file
Adds a new directory inside `bedrock/stories` - all articles for the blog will be under: `mozilla.org/stories/<slug>`

Very sorry about how large some of the scss files are - I still want to trim some before launch on 8/2

## Issue / Bugzilla link
#11634 

## Testing

Demo server URL: localhost:8000/stories

### CODE REVIEW

File: media/css/stories/includes/lib.scss [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+
+@font-face {
+    font-family: Antonio;
+    font-weight: 700;
+    font-style: normal;
+    font-display: swap;
+    src:
+        url('/media/fonts/fx-lifestyle/antonio-bold-webfont.woff2') format('woff2'),
+        url('/media/fonts/fx-lifestyle/antonio-bold-webfont.woff') format('woff');
+}
```
Comment (alexgibson) [inline media/css/stories/includes/lib.scss:32]: Heads up I'm removing these fonts and the page they were associated with in https://github.com/mozilla/bedrock/pull/11951

If you want to keep the fonts for this new page, I'd suggest moving them from `/fonts/fx-lifestyle/` to `/fonts/stories/`

Comment (nathan-barrett) [reply media/css/stories/includes/lib.scss:32]: Ok, thanks for the notice!

File: bedrock/stories/templates/stories/stories-base-article.html [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+                {% include 'includes/protocol/navigation/menus/innovation.html' %}
+              </ul>
+            </nav>
+          </div><!-- close .c-navigation-menu -->
+        </div><!-- close .c-navigation-items -->
+      </div><!-- close .c-navigation-container -->
+    </div><!-- close .c-navigation-l-content -->
+    <nav aria-label="breadcrumb" class="mzp-c-breadcrumb mzp-is-sticky">
+      <ol class="mzp-c-breadcrumb-list">
+        <li class="mzp-c-breadcrumb-item"><a href="{{ url('stories.landing') }}">Stories</a></li>
```
Comment (maureenlholland) [inline bedrock/stories/templates/stories/stories-base-article.html:62]: **issue (accessibility):** getting some contrast errors on the white text/pink background breadcrumb
Screen Shot 2022-07-28 at 1 05 14 PM

**suggestion:** this would be such a cool use case for the new[ color-contrast() function](https://developer.mozilla.org/en-US/docs/Web/CSS/color_value/color-contrast), but it's not supported yet. It's tricky because these backgrounds will change, but maybe keeping the black text of the heading is OK? likely need to check in with designer

File: bedrock/stories/templates/stories/articles/art-of-engagement.html [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+  <p>The goal isn’t to define specific solutions, but to let the ideas fly. “We’re asking the community to come up with the most innovative and blue-sky creative features that they can possibly think of that address a problem we’re trying to solve,” said Fembro. “Sometimes you'll see a really thorough post that clearly identifies a scenario or opportunity gap, and outlines a possible solution for us to consider.”</p>
+  <p>Once a feature has been incorporated into the browser, the Connect community can offer valuable feedback on whether it is performing as intended “in the wild,” Fembro said.</p>
+  <p>The team needed to discuss another, less tangible issue: how could they keep conversations from getting derailed? “Mozilla has built a rich culture over the years and from the beginning has been people-focused,” said Fembro. “A lot of our loyal users are very entrenched into our development process, which comes along with a lot of opinions.”</p>
+  <div class="c-stories-article-float-container left">
+    {{ stories_image(
+      image_url="img/stories/placeholder.png",
+      alt="",
```
Comment (maureenlholland) [inline bedrock/stories/templates/stories/articles/art-of-engagement.html:57]: **issue (accessibility):** once we get final images, it would be nice to have some descriptive alt text

File: bedrock/stories/templates/stories/stories-base-article.html [hunk 4]
```diff
@@ -0,0 +1,136 @@
+{#
+ This Source Code Form is subject to the terms of the Mozilla Public
+ License, v. 2.0. If a copy of the MPL was not distributed with this
+ file, You can obtain one at https://mozilla.org/MPL/2.0/.
+#}
+
+{% extends "base-protocol.html" %}
```
Comment (maureenlholland) [inline bedrock/stories/templates/stories/stories-base-article.html:7]: **issue (not blocking but high priority):**  I see Stories- Mozilla as page title on individual article pages and I think we want specific article titles, similarly, I wonder if we need more [meta data blocks ](https://github.com/mozilla/bedrock/blob/main/bedrock/base/templates/base-protocol.html#L26) to create the expected link previews when shared

**Review by maureenlholland — CHANGES_REQUESTED** [review]
There is so much great work here! 🎉 Very cool

Most of these comments are minor or non-blocking but a few are blocking. My primary concerns are some markup confusion, missing hrefs, and questionable meta data. I didn't do a close review of css code.

File: bedrock/stories/templates/stories/articles/art-of-engagement.html [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+{% endblock %}
+
+{% block article_content %}
+<section class="c-stories-article-section">
+  <p>Ray Fembro began by asking for feedback about all things regarding information organization. The project manager then got more specific: what thoughts did people have about tab management?</p>
+  <p>Plenty, as it turned out. Every day for the next two weeks, Fembro signed in to Connect, Mozilla’s new community platform, to listen. He reviewed comments, dissected feedback and otherwise engaged with Mozilla’s user community—one known for its talent, knowledge and an enthusiasm sometimes bordering on fervor.</p>
+  <p>With the launch of Connect, Mozilla has brought that community more directly into conversations with the employees in the midst of building the products they love. The platform acts as a collaborative space where people share product feedback, submit ideas, and have the kind of meaningful discussions that can impact future releases. “We’re opening up new opportunities for us to connect more with our users and make them more a part of product development,” said Product Community Director Jon Siddoway.</p>
```
Comment (maureenlholland) [inline bedrock/stories/templates/stories/articles/art-of-engagement.html:29]: feels like we should definitely have a link to Connect somewhere in here: https://connect.mozilla.org/
@slightlyoffbeat are there UTM params expected for links from the stories blog articles

File: bedrock/stories/templates/stories/stories-base-article.html [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+                {% include 'includes/protocol/navigation/menus/products.html' %}
+                {% include 'includes/protocol/navigation/menus/whoweare.html' %}
+                {% include 'includes/protocol/navigation/menus/innovation.html' %}
+              </ul>
+            </nav>
+          </div><!-- close .c-navigation-menu -->
+        </div><!-- close .c-navigation-items -->
+      </div><!-- close .c-navigation-container -->
+    </div><!-- close .c-navigation-l-content -->
+    <nav aria-label="breadcrumb" class="mzp-c-breadcrumb mzp-is-sticky">
```
[the remark this answers is not in the document]
Comment (nathan-barrett) [reply bedrock/stories/templates/stories/stories-base-article.html:60]: As far as I could tell, the `subnav` needs to be nested within the nav itself to be able to use the sticky-ness. I tried to use the same JS logic that the main nav uses (but have it be a sibling of the nav and not a child) and the animation seemed pretty janky.

[the remark this answers is not in the document]
Comment (nathan-barrett) [reply bedrock/stories/templates/stories/stories-base-article.html:60]: But would be happy to be able to move it outside of the main nav and not have to copy all of that markup

File: bedrock/stories/templates/stories/stories-base-article.html [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+
+
+{% block content %}
+  <article class="c-article-wrapper">
+    <section class="c-stories-article-header {% if self.article_hero() %}contains-hero{% endif %}">
+      <div class="c-stories-article-header-container">
+        <p class="is-featured">{% if featured %}Featured Article{% endif%}
+        <h1 class="c-stories-article-title">{% block article_title %}{% endblock %}</h1>
+        <aside class="c-stories-article-data">
+            <p class="grid-label">published</p>
```
[the remark this answers is not in the document]
Comment (nathan-barrett) [reply bedrock/stories/templates/stories/stories-base-article.html:77]: Nice! TY I was struggling to figure out how to make this accessible!

File: bedrock/stories/templates/stories/stories-base-article.html (hunk 6 above)
[the remark this answers is not in the document]
Comment (maureenlholland) [reply bedrock/stories/templates/stories/stories-base-article.html:60]: Yeah, I was wondering why the main nav was in here and then saw the fancy breadcrumb behaviour. Thanks for confirming!

File: bedrock/stories/templates/stories/stories-base-article.html [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+            {% endfor %}
+              </div>
+            </aside>
+        {% endif %}
+        <h2 class="c-stories-article-subtitle">{% block article_subtitle %}{% endblock %}</h2>
+        {% block article_content %}{% endblock %}
+      </div>
+      <section class="c-stories-social-share">
+        <ul class="social-share-list">
+          <li class="social-share-item"><a class="twitter">Twitter</a></li>
```
[the remark this answers is not in the document]
Comment (nathan-barrett) [reply bedrock/stories/templates/stories/stories-base-article.html:110]: Still waiting on the right url info for these so its a bit of a block at the moment!

File: bedrock/stories/templates/stories/stories-base-article.html (hunk 4 above)
Comment (nathan-barrett) [reply bedrock/stories/templates/stories/stories-base-article.html:7]: Good call out! I'll add more meta data as I get them!

**Comment by craigcook** [comment]
These images should be the same height so things line up horizontally. Or are these placeholders and you're still waiting for the final images?
image

File: media/css/stories/includes/lib.scss [hunk 9]
```diff
@@ -0,0 +1,202 @@
...
+    src:
+        url('/media/fonts/stories/source-serif/SourceSerifPro-Regular.woff2') format('woff2'),
+        url('/media/fonts/stories/source-serif/SourceSerifPro-Regular.woff') format('woff');
+    font-weight: normal;
+    font-style: normal;
+    font-display: swap;
+}
+
+@font-face {
+    font-family: 'Source Serif Pro';
```
Comment (craigcook) [inline media/css/stories/includes/lib.scss:168]: This is a lot of fonts. Are we really using all of these weights and styles? We should reduce it to just regular and bold. Maybe a regular italic and bold italic if they're actually used anywhere but I haven't noticed any italics.

**Review by craigcook — CHANGES_REQUESTED** [review]

File: bedrock/stories/templates/stories/articles/art-of-engagement.html [hunk 10]
```diff
@@ -0,0 +1,120 @@
...
+  <p>Plenty, as it turned out. Every day for the next two weeks, Fembro signed in to Connect, Mozilla’s new community platform, to listen. He reviewed comments, dissected feedback and otherwise engaged with Mozilla’s user community—one known for its talent, knowledge and an enthusiasm sometimes bordering on fervor.</p>
+
+  {{ stories_article_blockquote(
+    quote="We’re opening up new opportunities for us to connect more with our users",
+    caption="Siddoway"
+  )}}
+
+  <p class="bold">With the launch of Connect, Mozilla has brought that community more directly into conversations with the employees in the midst of building the products they love. The platform acts as a collaborative space where people share product feedback, submit ideas, and have the kind of meaningful discussions that can impact future releases. “We’re opening up new opportunities for us to connect more with our users and make them more a part of product development,” said Product Community Director Jon Siddoway.</p>\
```
Comment (maureenlholland) [inline bedrock/stories/templates/stories/articles/art-of-engagement.html:38]:
```suggestion
  <p class="bold">With the launch of Connect, Mozilla has brought that community more directly into conversations with the employees in the midst of building the products they love. The platform acts as a collaborative space where people share product feedback, submit ideas, and have the kind of meaningful discussions that can impact future releases. “We’re opening up new opportunities for us to connect more with our users and make them more a part of product development,” said Product Community Director Jon Siddoway.</p>
```

**Review by maureenlholland — COMMENTED** [review]
👍 LGTM, with two remaining minor issues:

1. We use the same related stories on each article (even if you are already reading one of the "related" articles, i.e. http://localhost:8000/en-US/stories/community-champion/). Ideally, I think we should rotate these so you can get to The Art of Engagement from the related stories section too.

2. Spacing is still a bit funky around blockquotes on large phone/small tablet screen sizes.
Screen Shot 2022-07-29 at 12 05 34 PM

File: bedrock/stories/templates/stories/includes/macros.html [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+    {{ image(
+      url=image_url,
+      alt=alt,
+      width=width,
+      height=height,
+      loading=loading,
+      include_highres=include_highres
+    ) }}
+    {% if caption %}
+      <figcaption>{{ caption }}</figcaption>
```
Comment (craigcook) [inline bedrock/stories/templates/stories/includes/macros.html:43]: This is pretty hard to read on the pink:
image

If we want a common style that works on different colored backgrounds (but isn't solid black) we could use an rgba color. Eventually we'll likely want a dark theme version of the article header too but that's a problem for the future.

**Review by craigcook — CHANGES_REQUESTED** [review]

File: bedrock/stories/templates/stories/stories-base-article.html [hunk 12]
```diff
@@ -0,0 +1,138 @@
...
+          <li class="social-share-item"><a class="pocket">Pocket</a></li>
+        </ul>
+      </section>
+      <section class="c-stories-related">
+        <div class="related-stories-grid">
+          <h2 class="c-stories-related-header">Related Stories</h2>
```
[the remark this answers is not in the document]
Comment (nathan-barrett) [reply bedrock/stories/templates/stories/stories-base-article.html]: Yeah - my idea that this was going to be using contentful to get the list of articles to render  - will need to be updated once the move to contentful happens

File: bedrock/stories/templates/stories/articles/art-of-engagement.html [hunk 13]
```diff
@@ -0,0 +1,120 @@
...
+  <p>The conversation progressed, with Siddoway and the managers agreeing that the community’s engagement would be most valuable to their work at two key stages of product development. At the start, when they’re brainstorming ideas for features and products, and at the end, when the developers are testing what they’ve built.</p>
+
+  {{ stories_article_blockquote(
+    quote="We are interested in how people use Firefox and how they think about their day-to-day browsing needs.",
+    caption="Safko",
+    class="color-accent"
+  )}}
+
+  <p class="bold">“It’s really useful for us to engage with users in exploratory situations where we know we want to improve a certain experience but we haven’t made a decision about the exact solution,” Safko said. “We are interested in hearing how people use something or how they think about the underlying need so we can better understand them.”</p>
```
[the remark this answers is not in the document]
Comment (nathan-barrett) [reply bedrock/stories/templates/stories/articles/art-of-engagement.html]: Good point - another ask from design. I'll see what they have to say - but I'll most likely put this on the back burner for now.

File: bedrock/stories/templates/stories/articles/art-of-engagement.html [hunk 14]
```diff
@@ -0,0 +1,148 @@
...
+  {"name": "Jon Siddoway", "title": "Community Manager", "image": "img/stories/credit/jon-siddoway-credit.png"},
+] %}
+
+{% set related = [
+   {
+    "title": "Licensed to Build Together",
+    "desc": "Who really owns the Firefox browser? Its copyleft license means we all do.",
+    "image_url": "img/stories/build-together.png",
+    "link_url": url('stories.articles.build-together'),
+    "image_alt": "Someone placing wooden blocks together"
```
Comment (craigcook) [inline bedrock/stories/templates/stories/articles/art-of-engagement.html:31]: I'm not sure how helpful it is to include alt text for these images, since they seem like kind of unrelated stock images barely connected to the content.

File: bedrock/stories/templates/stories/articles/art-of-engagement.html [hunk 15]
```diff
@@ -0,0 +1,148 @@
...
+  },
+  {
+    "title": "From Volunteer to Mozilla Community Champion",
+    "desc": "An early Mozilla volunteer's curiosity about open-source software led him to a career with a company whose mission he believes in.",
+    "image_url": "img/stories/community-champion.png",
+    "link_url": url('stories.articles.community-champion'),
+    "image_alt": "Mozilla Product Manager, Asa Dotzler"
```
Comment (craigcook) [inline bedrock/stories/templates/stories/articles/art-of-engagement.html:38]: If we do want to keep alt text, this should be updated to match the new image.

File: bedrock/stories/templates/stories/articles/art-of-engagement.html [hunk 16]
```diff
@@ -0,0 +1,148 @@
...
+    "title": "Licensed to Build Together",
+    "desc": "Who really owns the Firefox browser? Its copyleft license means we all do.",
+    "image_url": "img/stories/build-together.png",
+    "link_url": url('stories.articles.build-together'),
+    "image_alt": "Someone placing wooden blocks together"
+  },
+  {
+    "title": "From Volunteer to Mozilla Community Champion",
+    "desc": "An early Mozilla volunteer's curiosity about open-source software led him to a career with a company whose mission he believes in.",
+    "image_url": "img/stories/community-champion.png",
```
Comment (craigcook) [inline bedrock/stories/templates/stories/articles/art-of-engagement.html:36]: I think this image (and probably some others but definitely this one) would be much smaller as a JPEG. I can convert them if you need help.

File: bedrock/stories/templates/stories/articles/art-of-engagement.html [hunk 17]
```diff
@@ -0,0 +1,148 @@
...
+ file, You can obtain one at https://mozilla.org/MPL/2.0/.
+#}
+
+{% extends "stories/stories-base-article.html" %}
+
+{% from "stories/includes/macros.html" import stories_article_blockquote, stories_image with context %}
+
+
+{% block publish_date %}2022-08-02{% endblock %}
+{% block publish_date_formatted %}August, 2, 2022{% endblock %}
```
Comment (craigcook) [inline bedrock/stories/templates/stories/articles/art-of-engagement.html:13]: nit: No comma after the month.
```suggestion
{% block publish_date_formatted %}August 2, 2022{% endblock %}
```

File: bedrock/stories/templates/stories/articles/art-of-engagement.html [hunk 18]
```diff
@@ -0,0 +1,148 @@
...
+   {
+    "title": "Licensed to Build Together",
+    "desc": "Who really owns the Firefox browser? Its copyleft license means we all do.",
+    "image_url": "img/stories/build-together.png",
+    "link_url": url('stories.articles.build-together'),
+    "image_alt": "Someone placing wooden blocks together"
+  },
+  {
+    "title": "From Volunteer to Mozilla Community Champion",
+    "desc": "An early Mozilla volunteer's curiosity about open-source software led him to a career with a company whose mission he believes in.",
```
Comment (craigcook) [inline bedrock/stories/templates/stories/articles/art-of-engagement.html:35]: Nit: curly apostrophe.
```suggestion
    "desc": "An early Mozilla volunteer’s curiosity about open-source software led him to a career with a company whose mission he believes in.",
```

File: bedrock/stories/templates/stories/articles/art-of-engagement.html [hunk 19]
```diff
@@ -0,0 +1,148 @@
...
+      include_highres=True,
+      alt="illustration of a brain in a jar",
+      width="1010",
+      height="519",
+      loading="lazy",
+      image_class="article-width center",
+      caption="Image by: Opensource.com / CC BY-SA 4.0"
+    )}}
+
+    <p>At the core of the 1998 Mozilla project was a commitment to open-source development, (Link to License story) and the volunteers who believed in that model played a vital role in developing Firefox 1.0, released six years later. Since then, thousands of people have offered ideas, suggestions and criticisms that continue to shape the browser and other Mozilla products. Between 2017 and 2019 alone, 22,821 non-employees made more than 2 million contributions to Mozilla products, according to the 2020 report “Mozilla and the Rebel Alliance.” Almost 750,000 of those contributions focused on Firefox, with more than 65,000 specifically involving code.</p>
```
Comment (craigcook) [inline bedrock/stories/templates/stories/articles/art-of-engagement.html:94]: Missing link:
```suggestion
    <p>At the core of the 1998 Mozilla project was a <a href="{{ url('stories.build-together') }}">commitment to open-source development</a>, and the volunteers who believed in that model played a vital role in developing Firefox 1.0, released six years later. Since then, thousands of people have offered ideas, suggestions and criticisms that continue to shape the browser and other Mozilla products. Between 2017 and 2019 alone, 22,821 non-employees made more than 2 million contributions to Mozilla products, according to the 2020 report “Mozilla and the Rebel Alliance.” Almost 750,000 of those contributions focused on Firefox, with more than 65,000 specifically involving code.</p>
```

File: bedrock/stories/templates/stories/articles/build-together.html [hunk 20]
```diff
@@ -0,0 +1,67 @@
...
+    "desc": "An early Mozilla volunteer's curiosity about open-source software led him to a career with a company whose mission he believes in.",
+    "image_url": "img/stories/community-champion.png",
+    "link_url": url('stories.articles.community-champion'),
+    "image_alt": "Mozilla Product Manager, Asa Dotzler"
+  }
+] %}
+
+
+{% block publish_date %}2022-08-02{% endblock %}
+{% block publish_date_formatted %}August, 2, 2022{% endblock %}
```
Comment (craigcook) [inline bedrock/stories/templates/stories/articles/build-together.html:38]:
```suggestion
{% block publish_date_formatted %}August 2, 2022{% endblock %}
```

File: bedrock/stories/templates/stories/articles/community-champion.html [hunk 21]
```diff
@@ -0,0 +1,74 @@
...
+    "desc": "Mozilla’s vibrant community gets a direct connection to the people building the products they love.",
+    "image_url": "img/stories/art-of-engagement.png",
+    "link_url": url('stories.articles.art-of-engagement'),
+    "image_alt": "Image of a brain in a jar"
+  }
+] %}
+
+
+{% block publish_date %}2022-08-02{% endblock %}
+{% block publish_date_formatted %}August, 2, 2022{% endblock %}
```
Comment (craigcook) [inline bedrock/stories/templates/stories/articles/community-champion.html:36]:
```suggestion
{% block publish_date_formatted %}August 2, 2022{% endblock %}
```

File: bedrock/stories/templates/stories/articles/community-champion.html [hunk 22]
```diff
@@ -0,0 +1,74 @@
...
+
+{% block article_title %}From Volunteer to Mozilla Community Champion{% endblock %}
+{% block article_subtitle %}Asa Dotzler's curiosity about open-source software led him to a career with a company whose mission he believes in.{% endblock %}
```
Comment (craigcook) [inline bedrock/stories/templates/stories/articles/community-champion.html:39]:
```suggestion
{% block article_subtitle %}Asa Dotzler’s curiosity about open-source software led him to a career with a company whose mission he believes in.{% endblock %}
```

File: bedrock/stories/templates/stories/includes/macros.html [hunk 23]
```diff
@@ -0,0 +1,55 @@
...
+    {% endif %}
+  </figure>
+{%- endmacro %}
+
+{% macro stories_article_blockquote(quote="", caption="", class="") %}
+  <figure class="c-stories-article-blockquote {{ class }}">
+   <blockquote class="{% if caption %}contains caption{% endif %}">“{{ quote }}”</blockquote>
```
Comment (craigcook) [inline bedrock/stories/templates/stories/includes/macros.html:49]: Is this class used at this point? Is there a reason it's two classes or should this be something like `contains-caption`? (or `has-caption` if we follow our usual naming convention).

If we don't need it now we can remove it, but if it's still useful the if could wrap around the whole attribute like `{% if caption %}class="has-caption"{% endif %}`

**Review by craigcook — CHANGES_REQUESTED** [review]
Mostly just nitpicks and some tidying, no real show-stoppers except the one missing link (I'll do another readthrough in case I missed any more).

File: bedrock/stories/templates/stories/articles/art-of-engagement.html (hunk 14 above)
Comment (nathan-barrett) [reply bedrock/stories/templates/stories/articles/art-of-engagement.html:31]: heh, probably right!

**Review by craigcook — APPROVED** [review]
